CIP-onchain-cert-metadata/README.md

```diff
@@ -0,0 +1,489 @@
+---
+CIP: ?
+Title: On-chain Certification Metadata Standard
+Category: Metadata
+Status: Proposed
+Authors:
+    - Romain Soulat <romain.soulat@iohk.io>
+    - Nicolas Jeannerod <nicolas.jeannerod@tweag.io>
+    - Mathieu Montin <mathieu.montin@tweag.io>
+    - M. Ali Modiri <mixaxim@gimbalabs.io>
+    - Simon Thompson <simon.thompson@iohk.io>
+Implementors: []
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/499
+    - https://github.com/input-output-hk/Certification-working-group/pull/18
+    - https://docs.google.com/document/d/1vC4cS-gMLIvgduSh5Fipn_lU_0Ou0NY4X_a9lWJz2B8
+Created: 2023-02-23
+License: CC-BY-4.0
+---
+
+
+## Abstract
+
+Certification is the process of providing various kinds of assurance of DApps through the provision of verifiable, trustworthy metadata linking to relevant evidence of assurance, such as audit reports, test run data, formal verification proofs etc.
+
+Currently, certification does not have a standardised way to be stored and presented to all stakeholders (DApp developer, DApp stores, end-users, auditors, ...) in an immutable, persistent and verifiable way.
+
+This CIP descibes a standardised method for certificates to be published and stored on-chain and for stakeholders to be able to verify the different claims of this certificate.
+
+This proposal describes how certification metadata can be designed for DApp registration. It first describes the requirements for DApp certification, and then sets out the options for its inclusion in CIP-72.
+
+## Motivation: why is this CIP necessary?
+
+It is expected that evidence of various kinds of assurance of DApps is recorded in an immutable and verifiable way on the Cardano blockchain; these forms of assurance are: automated testing, including property-based testing (level 1), audit (level 2) and formal verification (level 3).
+
+The metadata should be discoverable by all certification stakeholders, including end-users, DApp developers, and ecosystem components, such as light wallets and DApp stores. Information should be indexable by certification issuer, DApp developer, DApp and DApp version.
+
+### Use-cases and Stakeholders
+
+DApp developers will seek certification from one of the issuers, according to their desired level of certification and will refer to the certificate on several platforms (e.g. on their website or in their DApp registration on a DApp store).
+
+Certification is to be provided by certification issuers including: testing services (level 1), auditors (level 2), and verification services (level 3). Certification issuers will issue these certificates referring to a particular version of a DApp that has succesfully gone through a certification process.
```

Comment:
The levels of certification, as mentioned in the paragraph, are defined based on the type of certification issuer and the level of testing and verification involved.

Level 1 certification is provided by testing laboratories, which typically involve automated and manual testing to ensure the DApp meets certain functional and security requirements.

Level 2 certification is provided by Cardano's auditors, who perform a more detailed review of the DApp's code, architecture, and security, to ensure that it meets higher standards.

Level 3 certification is provided by Cardano certification issuers and verification services, which typically involve a more comprehensive review of the DApp's design, implementation, and security, and may include formal verification methods to mathematically prove the correctness of the DApp's behavior.

In terms of the differences between auditors and verifiers, auditors typically focus on reviewing the code and implementation of the DApp, while verifiers may also take into account the overall design and architecture of the DApp. Additionally, verification may involve formal methods such as model checking, theorem proving or static analysis are commonly used in verification to rigorously verify the correctness of a DApp's behavior. For example, model checking involves systematically exploring all possible states of a DApp to check if it satisfies certain properties, while theorem proving involves using mathematical logic to prove that a DApp behaves correctly under all possible circumstances. These formal methods are typically not used in auditing, which may rely more on manual review and testing.

Reply:
@MIxAxIM 

Thank you for this very well written answer. Perhaps its my lack of understanding - and these are common well known standards?

If these are well known specifications, then linking to them would avoid such questions from less knowledgeable people such as myself. Or if these are levels being defined here then IMO it would be best to include such detail as you have in this answer into the proposal.

Reply:
Exactly, they refer to those certification levels:
https://iohk.io/en/blog/posts/2021/10/25/new-certification-levels-for-smart-contracts-on-cardano/

They are yet to be defined properly and we've even refined their definition (in an informal way) during the working group meetings. I can add a reference to a document and add that the actual levels will be standardised in a near future.

Reply:
@RSoulatIOHK yes I think it would be a great idea to add this link.  If you added it somewhere please let us know because I can't find it.

Reply:
resolving: I can see this is elaborated in the revised text:
> Three levels of certications [sic; I'll submit a correction for this] lare thought of at the time of this CIP: level 1 to 3. They were presented at the following [link](https://iohk.io/en/blog/posts/2021/10/25/new-certification-levels-for-smart-contracts-on-cardano/) and will be fully described in a future document.